first-read-last-write-cache/src/cache.rs

```diff
@@ -427,4 +393,88 @@ mod tests {
             }
         }
     }
+
+    proptest! {
+        #[test]
+        fn test_merge_fuzz(s: u8) {
+            let mut testvec = vec![0; 15];
+            for i in 0..15 {
+                testvec[i] = ((s as u16 + i as u16) % 255) as u8;
```

Comment:
Two small nits:
1. You can make `s: usize` and remove `s as u16` cast
2. Let's extract 15 a variable in this function.